# Proyek Analisis Data: oil-and-gas-summary-production-data-1967
- **Nama:** Diftha Ariangga Firdaus
- **Email:** difthadaus@gmail.com

## Menentukan Pertanyaan Bisnis

1. Bagaimana tren produksi minyak dan gas per tahun selama periode analisis?

2. County mana yang memiliki produksi minyak dan gas tertinggi selama periode analisis?

3. Operator mana yang memiliki produksi minyak dan gas tertinggi selama periode analisis?

4. Bagaimana hubungan antara jumlah sumur aktif dengan produksi minyak dan gas?

5. Bagaimana distribusi geografis produksi minyak dan gas berdasarkan lokasi sumur selama periode analisis?

## Import Semua Packages/Library yang Digunakan

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import folium

## Data Wrangling

### Gathering Data

Memuat data oil dan gas

In [2]:
data = pd.read_csv("oil-and-gas-summary-production-data-1967.csv",sep=",")
data.head()

,Production Year,Production Date Entered,Operator,County,Town,Field,Producing Formation,Active Oil Wells,Inactive Oil Wells,Active Gas Wells,Inactive Gas Wells,Injection Wells,Disposal Wells,Self-use Well,"Oil Produced, bbl","Gas Produced, Mcf","Water produced, bbl","Taxable Gas, Mcf",Purchaser Codes,Location
0,1995,12/3/1996,"Buffalo China, Inc.",Erie,BUFFALO,BUFFALO,MEDINA,0,0,1,0,0,0,YES,0,106,0,0,NaN,"BUFFALO, NY\n(42.887691, -78.879374)"
1,1995,3/29/1996,"Copper Ridge Oil, Inc.",Steuben,WEST UNION,BEECH HILL-INDEPENDENCE,FULMER VALLEY,28,0,0,0,7,0,NO,1229,0,180,0,OA,"WEST UNION, NY\n(42.094951, -77.725816)"
2,1994,2/14/1995,"White, Walter W. & Christina L.",Cattaraugus,CARROLLTON,BRADFORD,BRADFORD,3,0,0,0,0,0,NO,462,0,0,0,OA,"CARROLLTON, NY\n(42.023289, -78.628438)"
3,1994,1/19/1995,"Stiegler, Richard M",Erie,CHEEKTOWAGA,BUFFALO,MEDINA,0,0,1,0,0,0,YES,0,530,0,0,NaN,"CHEEKTOWAGA, NY\n(42.907071, -78.754318)"
4,1995,2/12/1996,"Bucher, Charles J",Cattaraugus,ALLEGANY,FIVE MILE,BRADFORD,10,0,0,0,0,0,NO,45,0,80,0,OA,"ALLEGANY, NY\n(42.088061, -78.491258)"


**Insight**
Dataset yang dimuat berisi informasi mengenai aktivitas produksi minyak dan gas dari berbagai sumur yang dioperasikan oleh beberapa perusahaan atau operator. Setiap baris data merepresentasikan satu entri produksi yang mencakup informasi seperti tahun produksi, tanggal data dimasukkan, nama operator, serta lokasi sumur yang terdiri dari county, town, field, dan koordinat lokasi. Selain itu, dataset juga memuat informasi mengenai jumlah dan status sumur, seperti jumlah sumur minyak aktif dan tidak aktif, sumur gas aktif dan tidak aktif, serta sumur injeksi dan disposal. Data produksi yang dicatat meliputi jumlah minyak yang diproduksi (barrel), gas yang diproduksi (Mcf), air yang dihasilkan, serta gas yang dikenakan pajak. Dataset ini juga memiliki atribut Self-use Well yang menunjukkan apakah sumur tersebut digunakan untuk keperluan sendiri atau tidak. Secara keseluruhan, dataset ini memberikan gambaran mengenai kondisi operasional sumur serta hasil produksi minyak dan gas pada berbagai lokasi dan operator.

### Assessing Data

Pemeriksaan struktur data oil and gas

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30053 entries, 0 to 30052
Data columns (total 20 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Production Year          30053 non-null  int64 
 1   Production Date Entered  30053 non-null  object
 2   Operator                 30053 non-null  object
 3   County                   30022 non-null  object
 4   Town                     29396 non-null  object
 5   Field                    28772 non-null  object
 6   Producing Formation      29393 non-null  object
 7   Active Oil Wells         30053 non-null  int64 
 8   Inactive Oil Wells       30053 non-null  int64 
 9   Active Gas Wells         30053 non-null  int64 
 10  Inactive Gas Wells       30053 non-null  int64 
 11  Injection Wells          30053 non-null  int64 
 12  Disposal Wells           30053 non-null  int64 
 13  Self-use Well            29434 non-null  object
 14  Oil Produced, bbl        30053 non-nul

Pemeriksaan missing value 

In [4]:
data.isna().sum()

Production Year                0
Production Date Entered        0
Operator                       0
County                        31
Town                         657
Field                       1281
Producing Formation          660
Active Oil Wells               0
Inactive Oil Wells             0
Active Gas Wells               0
Inactive Gas Wells             0
Injection Wells                0
Disposal Wells                 0
Self-use Well                619
Oil Produced, bbl              0
Gas Produced, Mcf              0
Water produced, bbl            0
Taxable Gas, Mcf               0
Purchaser Codes            11798
Location                       0
dtype: int64

Pemeriksaan data duplikasi

In [5]:
print("Jumlah duplikasi: ", data.duplicated().sum())

Jumlah duplikasi:  4


Analisis statistik deskriptif data

In [6]:
data.describe(include='all')

,Production Year,Production Date Entered,Operator,County,Town,Field,Producing Formation,Active Oil Wells,Inactive Oil Wells,Active Gas Wells,Inactive Gas Wells,Injection Wells,Disposal Wells,Self-use Well,"Oil Produced, bbl","Gas Produced, Mcf","Water produced, bbl","Taxable Gas, Mcf",Purchaser Codes,Location
count,30053.000000,30053,30053,30022,29396,28772,29393,30053.000000,30053.000000,30053.000000,30053.000000,30053.000000,30053.000000,29434,30053.000000,3.005300e+04,3.005300e+04,3.005300e+04,18255,30053
unique,NaN,1257,1771,30,250,229,63,NaN,NaN,NaN,NaN,NaN,NaN,3,NaN,NaN,NaN,NaN,130,251
top,NaN,9/20/1994,National Fuel Gas Supply Corp.,Chautauqua,ALLEGANY,LAKESHORE,MEDINA,NaN,NaN,NaN,NaN,NaN,NaN,NO,NaN,NaN,NaN,NaN,GK,"ALLEGANY, NY\n(42.088061, -78.491258)"
freq,NaN,272,974,9137,1514,8557,17938,NaN,NaN,NaN,NaN,NaN,NaN,18325,NaN,NaN,NaN,NaN,6297,1514
mean,1987.929225,NaN,NaN,NaN,NaN,NaN,NaN,5.040262,1.391209,3.724553,0.600106,1.270489,0.044821,NaN,811.933118,1.811270e+04,6.251382e+03,9.025647e+03,NaN,NaN
std,7.935213,NaN,NaN,NaN,NaN,NaN,NaN,25.146264,8.449948,12.362509,2.449830,16.456011,5.093759,NaN,8263.100877,8.805514e+04,1.787794e+05,7.249590e+04,NaN,NaN
min,1967.000000,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,0.000000,0.000000e+00,0.000000e+00,0.000000e+00,NaN,NaN
25%,1983.000000,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,0.000000,0.000000e+00,0.000000e+00,0.000000e+00,NaN,NaN
50%,1989.000000,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,NaN,0.000000,3.210000e+02,0.000000e+00,0.000000e+00,NaN,NaN
75%,1994.000000,NaN,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,2.000000,0.000000,0.000000,0.000000,NaN,0.000000,7.206000e+03,0.000000e+00,0.000000e+00,NaN,NaN


**Insight**
Pada tahap assessing data, dilakukan pemeriksaan terhadap struktur, tipe data, dan kelengkapan nilai pada dataset. Berdasarkan hasil pemeriksaan menggunakan info() dan describe(), dataset memiliki 30.053 baris data dan 20 kolom yang terdiri dari 11 kolom bertipe numerik (int64) dan 9 kolom bertipe objek (object). Kolom numerik umumnya berisi informasi jumlah sumur serta jumlah produksi seperti Active Oil Wells, Active Gas Wells, Oil Produced, Gas Produced, dan Water Produced, sedangkan kolom objek berisi informasi kategorikal seperti Operator, County, Town, Field, Producing Formation, dan Location.

Selain itu, ditemukan beberapa kolom yang memiliki nilai kosong (missing values), seperti pada kolom County, Town, Field, Producing Formation, Self-use Well, dan terutama Purchaser Codes yang memiliki jumlah nilai kosong cukup besar dibandingkan kolom lainnya. Hal ini menunjukkan bahwa tidak semua informasi tercatat secara lengkap pada setiap entri data. Dari statistik deskriptif juga terlihat bahwa rentang tahun produksi berada antara 1967 hingga 1999, dengan rata-rata sekitar tahun 1987, yang menunjukkan bahwa dataset mencakup data produksi dalam periode yang cukup panjang. Selain itu, nilai maksimum produksi minyak dan gas menunjukkan adanya variasi produksi yang cukup besar antar sumur atau lokasi produksi. Temuan-temuan ini memberikan gambaran awal mengenai karakteristik dataset sebelum dilakukan analisis lebih lanjut.

### Cleaning Data

Mengisi Field dari Producing Formation

In [7]:
data["Field"] = data["Field"].fillna(data["Producing Formation"])

Mengisi County dari Location

In [8]:
data["County"] = data["County"].fillna(
    data["Location"].str.split(",").str[0]
)

Mengisi Town dari Location

In [9]:
data["Town"] = data["Town"].fillna(
    data["Location"].str.split(",").str[0]
)

Mengisi Producing Formation dari Field

In [10]:
data["Producing Formation"] = data["Producing Formation"].fillna(data["Field"])

Mengisi Self-use Well menjadi INACTIVE

In [11]:
data.loc[
    (data["Active Oil Wells"] == 0) &
    (data["Active Gas Wells"] == 0) &
    (data["Inactive Oil Wells"] == 0) &
    (data["Inactive Gas Wells"] == 0),
    "Self-use Well"
] = data["Self-use Well"].fillna("INACTIVE")

Mengisi Self-use Well menjadi NO

In [12]:
data.loc[
    (
        (data["Active Oil Wells"] > 0) |
        (data["Inactive Oil Wells"] > 0) |
        (data["Active Gas Wells"] > 0) |
        (data["Inactive Gas Wells"] > 0) |
        (data["Injection Wells"] > 0) |
        (data["Disposal Wells"] > 0)
    )
    & (data["Self-use Well"].isna()),
    "Self-use Well"
] = "NO"

Menghapus Baris yang Masih Kosong pada Field dan Producing Formation

In [13]:
data = data.dropna(subset=["Field", "Producing Formation"])

Menghapus Kolom Purchaser Codes

In [14]:
data = data.drop(columns=["Purchaser Codes"])

Mengubah kolom Production Date Entered menjadi tipe data datetime

In [15]:
data["Production Date Entered"] = pd.to_datetime(data["Production Date Entered"])

Menghapus data duplikat

In [16]:
data = data.drop_duplicates()

Mengecek info data dan statistik data setelah di cleaning

In [17]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 29556 entries, 0 to 30052
Data columns (total 19 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   Production Year          29556 non-null  int64         
 1   Production Date Entered  29556 non-null  datetime64[ns]
 2   Operator                 29556 non-null  object        
 3   County                   29556 non-null  object        
 4   Town                     29556 non-null  object        
 5   Field                    29556 non-null  object        
 6   Producing Formation      29556 non-null  object        
 7   Active Oil Wells         29556 non-null  int64         
 8   Inactive Oil Wells       29556 non-null  int64         
 9   Active Gas Wells         29556 non-null  int64         
 10  Inactive Gas Wells       29556 non-null  int64         
 11  Injection Wells          29556 non-null  int64         
 12  Disposal Wells           29556 non-nu

In [18]:
data.describe(include='all')

,Production Year,Production Date Entered,Operator,County,Town,Field,Producing Formation,Active Oil Wells,Inactive Oil Wells,Active Gas Wells,Inactive Gas Wells,Injection Wells,Disposal Wells,Self-use Well,"Oil Produced, bbl","Gas Produced, Mcf","Water produced, bbl","Taxable Gas, Mcf",Location
count,29556.000000,29556,29556,29556,29556,29556,29556,29556.000000,29556.000000,29556.000000,29556.000000,29556.000000,29556.000000,29556,29556.000000,2.955600e+04,2.955600e+04,2.955600e+04,29556
unique,NaN,NaN,1703,31,249,250,93,NaN,NaN,NaN,NaN,NaN,NaN,3,NaN,NaN,NaN,NaN,249
top,NaN,NaN,National Fuel Gas Supply Corp.,Chautauqua,ALLEGANY,LAKESHORE,MEDINA,NaN,NaN,NaN,NaN,NaN,NaN,NO,NaN,NaN,NaN,NaN,"ALLEGANY, NY\n(42.088061, -78.491258)"
freq,NaN,NaN,971,9080,1487,8556,17936,NaN,NaN,NaN,NaN,NaN,NaN,18513,NaN,NaN,NaN,NaN,1487
mean,1988.043883,1993-02-08 10:38:41.071863680,NaN,NaN,NaN,NaN,NaN,5.063676,1.380194,3.784375,0.602923,1.290432,0.045575,NaN,822.213696,1.841462e+04,6.035954e+03,9.175591e+03,NaN
min,1967.000000,1984-01-31 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,0.000000,0.000000e+00,0.000000e+00,0.000000e+00,NaN
25%,1983.000000,1989-07-24 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,0.000000,0.000000e+00,0.000000e+00,0.000000e+00,NaN
50%,1989.000000,1994-06-01 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,NaN,0.000000,3.990000e+02,0.000000e+00,0.000000e+00,NaN
75%,1995.000000,1996-01-16 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,2.000000,0.000000,0.000000,0.000000,NaN,0.000000,7.538000e+03,1.000000e+00,0.000000e+00,NaN
max,1999.000000,2002-04-09 00:00:00,NaN,NaN,NaN,NaN,NaN,1273.000000,360.000000,684.000000,118.000000,1108.000000,846.000000,NaN,675589.000000,4.827263e+06,1.759123e+07,4.827263e+06,NaN


**Insight**

Proses data cleaning dilakukan untuk meningkatkan kualitas dataset dengan menangani nilai yang hilang dan menghapus data yang tidak relevan. Nilai kosong pada kolom Field dan Producing Formation diisi dengan memanfaatkan hubungan antar kolom karena keduanya berkaitan dengan lokasi produksi, sementara nilai kosong pada County dan Town diisi menggunakan informasi dari kolom Location. Pada kolom Self-use Well, nilai kosong ditentukan berdasarkan kondisi aktivitas sumur, di mana jika semua sumur aktif dan tidak aktif bernilai nol maka dianggap INACTIVE, sedangkan jika terdapat aktivitas sumur maka diisi NO. Selain itu, sebanyak 492 baris data yang masih tidak memiliki informasi pada kolom Field dan Producing Formation dihapus karena kedua atribut tersebut penting untuk analisis lokasi produksi. Kolom Purchaser Codes juga dihapus karena memiliki jumlah missing value yang sangat besar dan tidak digunakan dalam analisis utama. Setelah proses ini dilakukan, dataset menjadi lebih bersih, konsisten, dan siap digunakan untuk analisis lebih lanjut seperti eksplorasi data dan visualisasi produksi minyak dan gas.

### Exploratory Data Analysis (EDA)

In [20]:
data.describe(include='all')

,Production Year,Production Date Entered,Operator,County,Town,Field,Producing Formation,Active Oil Wells,Inactive Oil Wells,Active Gas Wells,Inactive Gas Wells,Injection Wells,Disposal Wells,Self-use Well,"Oil Produced, bbl","Gas Produced, Mcf","Water produced, bbl","Taxable Gas, Mcf",Location
count,29556.000000,29556,29556,29556,29556,29556,29556,29556.000000,29556.000000,29556.000000,29556.000000,29556.000000,29556.000000,29556,29556.000000,2.955600e+04,2.955600e+04,2.955600e+04,29556
unique,NaN,NaN,1703,31,249,250,93,NaN,NaN,NaN,NaN,NaN,NaN,3,NaN,NaN,NaN,NaN,249
top,NaN,NaN,National Fuel Gas Supply Corp.,Chautauqua,ALLEGANY,LAKESHORE,MEDINA,NaN,NaN,NaN,NaN,NaN,NaN,NO,NaN,NaN,NaN,NaN,"ALLEGANY, NY\n(42.088061, -78.491258)"
freq,NaN,NaN,971,9080,1487,8556,17936,NaN,NaN,NaN,NaN,NaN,NaN,18513,NaN,NaN,NaN,NaN,1487
mean,1988.043883,1993-02-08 10:38:41.071863680,NaN,NaN,NaN,NaN,NaN,5.063676,1.380194,3.784375,0.602923,1.290432,0.045575,NaN,822.213696,1.841462e+04,6.035954e+03,9.175591e+03,NaN
min,1967.000000,1984-01-31 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,0.000000,0.000000e+00,0.000000e+00,0.000000e+00,NaN
25%,1983.000000,1989-07-24 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,NaN,0.000000,0.000000e+00,0.000000e+00,0.000000e+00,NaN
50%,1989.000000,1994-06-01 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,1.000000,0.000000,0.000000,0.000000,NaN,0.000000,3.990000e+02,0.000000e+00,0.000000e+00,NaN
75%,1995.000000,1996-01-16 00:00:00,NaN,NaN,NaN,NaN,NaN,0.000000,0.000000,2.000000,0.000000,0.000000,0.000000,NaN,0.000000,7.538000e+03,1.000000e+00,0.000000e+00,NaN
max,1999.000000,2002-04-09 00:00:00,NaN,NaN,NaN,NaN,NaN,1273.000000,360.000000,684.000000,118.000000,1108.000000,846.000000,NaN,675589.000000,4.827263e+06,1.759123e+07,4.827263e+06,NaN


Tren Produksi Minyak dan Gas per Tahun

In [38]:
data["Year"] = data["Production Date Entered"].dt.year

trend_production = data.groupby("Year")[["Oil Produced, bbl", "Gas Produced, Mcf"]].sum()

trend_production

,"Oil Produced, bbl","Gas Produced, Mcf"
Year,,
1984,752157,19432081
1985,944140,27405602
1986,967595,33112015
1987,762094,34816019
1988,645281,29492528
1989,480272,27571389
1990,497817,26099797
1991,387364,24725129
1992,392191,24801788


County dengan Produksi Tertinggi

In [46]:
county_production = data.groupby("County")[["Oil Produced, bbl", "Gas Produced, Mcf"]].sum()

county_production = county_production.sort_values("Oil Produced, bbl", ascending=False)

county_production.head(10)

,"Oil Produced, bbl","Gas Produced, Mcf"
County,,
Cattaraugus,15278451,44585574
Allegany,5129610,17231504
Chautauqua,3348422,288686763
Steuben,528723,30925188
Erie,15642,55881064
Genesee,500,24784465
Wyoming,0,12097968
Wayne,0,3625
Tompkins,0,0


Operator dengan Produksi Tertinggi

In [47]:
operator_production = data.groupby("Operator")[["Oil Produced, bbl", "Gas Produced, Mcf"]].sum()

operator_production = operator_production.sort_values("Oil Produced, bbl", ascending=False)

operator_production.head(10)

,"Oil Produced, bbl","Gas Produced, Mcf"
Operator,,
Pennzoil Products Co.,7536048,4659306
Witco Corp.,1426487,547382
"Ebenezer Oil Co., Inc.",1392305,20961
Kinley Oil & Gas Corp.,1132099,21700
Quaker State Corp.,897610,7897755
Richardson Petroleum Corp.,803987,1784069
K-Arcy Corp.,714750,0
Berea Oil & Gas Corp.,570711,9411766
Vosburg Oil Co.,411901,0


Hubungan Jumlah Sumur Aktif dengan Produksi (Membuat Total Sumur Aktif)

In [48]:
data["Total Active Wells"] = data["Active Oil Wells"] + data["Active Gas Wells"]

Membuat Cluster Manual

In [49]:
def cluster_well(x):
    if x == 0:
        return "Tidak Ada Sumur Aktif"
    elif x <= 5:
        return "Sumur Rendah"
    elif x <= 20:
        return "Sumur Sedang"
    else:
        return "Sumur Tinggi"

data["Well Cluster"] = data["Total Active Wells"].apply(cluster_well)

Melihat Produksi

In [50]:
cluster_production = data.groupby("Well Cluster")[["Oil Produced, bbl","Gas Produced, Mcf"]].mean()

cluster_production

,"Oil Produced, bbl","Gas Produced, Mcf"
Well Cluster,,
Sumur Rendah,113.155840,9006.457625
Sumur Sedang,680.187672,25612.489007
Sumur Tinggi,6551.891770,91408.232019
Tidak Ada Sumur Aktif,4.805414,212.927423


Mengambil Latitude dan Longitude dari Location

In [51]:

data["Latitude"] = data["Location"].str.extract(r'\((.*?),')[0]
data["Longitude"] = data["Location"].str.extract(r',\s(.*?)\)')[0]

data["Latitude"] = pd.to_numeric(data["Latitude"])
data["Longitude"] = pd.to_numeric(data["Longitude"])

Melihat Statistik Produksi Berdasarkan Lokasi

In [52]:
location_production = data.groupby(["Latitude","Longitude"])[["Oil Produced, bbl","Gas Produced, Mcf"]].sum()

location_production = location_production.sort_values(by="Oil Produced, bbl", ascending=False)

location_production.head()

,,"Oil Produced, bbl","Gas Produced, Mcf"
Latitude,Longitude,,
42.088061,-78.491258,9946555,1146408
42.023289,-78.628438,3840505,1973285
42.012207,-78.060759,1401876,1412502
42.038067,-79.283549,1262300,11113922
42.077495,-78.429849,1201482,4732316


**Insight**

Berdasarkan hasil Exploratory Data Analysis (EDA), terlihat bahwa produksi minyak dan gas mengalami fluktuasi selama periode analisis. Pada awal periode (sekitar tahun 1984–1993) produksi minyak berada pada kisaran ratusan ribu barel per tahun dan produksi gas mencapai puluhan juta Mcf, kemudian terjadi lonjakan produksi yang sangat tinggi pada tahun 1994 sebelum kembali menurun secara signifikan pada tahun-tahun berikutnya hingga mencapai tingkat yang sangat rendah pada awal tahun 2000-an. Dari sisi wilayah, County Cattaraugus tercatat sebagai daerah dengan produksi minyak tertinggi, sedangkan County Chautauqua memiliki produksi gas paling besar dibandingkan wilayah lainnya, menunjukkan adanya spesialisasi sumber daya di tiap wilayah. Analisis berdasarkan operator menunjukkan bahwa Pennzoil Products Co. merupakan operator dengan produksi minyak terbesar, sementara beberapa operator lain seperti Quaker State Corp. dan Berea Oil & Gas Corp. juga memiliki kontribusi signifikan terutama pada produksi gas. Hasil clustering manual berdasarkan jumlah sumur aktif menunjukkan hubungan yang jelas antara jumlah sumur dan tingkat produksi, di mana kelompok Sumur Tinggi menghasilkan rata-rata produksi minyak dan gas paling besar, sedangkan kelompok tanpa sumur aktif memiliki produksi yang sangat rendah. Selain itu, analisis koordinat geografis menunjukkan bahwa produksi energi terkonsentrasi pada beberapa titik lokasi tertentu, dengan beberapa koordinat seperti sekitar (42.088061, -78.491258) dan (42.023289, -78.628438) menjadi hotspot produksi minyak dan gas, yang mengindikasikan adanya area eksploitasi sumber daya yang lebih intensif dibandingkan wilayah lainnya.

## Visualization & Explanatory Analysis

Membuat DataFrame Gabungan (all_df)

In [53]:
data.to_csv("all_data.csv", index=False)